src/test/java/org/terasology/itempipes/ItemPipesTest.java

```diff
@@ -28,91 +32,105 @@
+import org.terasology.moduletestingenvironment.extension.UseWorldGenerator;
+import org.terasology.registry.In;
 import org.terasology.world.BlockEntityRegistry;
 import org.terasology.world.WorldProvider;
 import org.terasology.world.block.Block;
 import org.terasology.world.block.BlockManager;
-import org.terasology.world.block.entity.placement.PlaceBlocks;
+import org.terasology.world.block.BlockRegion;
 import org.terasology.world.block.family.BlockFamily;
+import org.terasology.world.block.family.BlockPlacementData;
 import org.terasology.world.block.items.BlockItemFactory;
-import org.terasology.world.block.items.OnBlockItemPlaced;
 
 import java.util.EnumSet;
-import java.util.Set;
 
 import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertTrue;
 
-public class ItemPipesTest extends ModuleTestingEnvironment {
+@ExtendWith(MTEExtension.class)
+@UseWorldGenerator("ModuleTestingEnvironment:empty")
+@Dependencies({"ItemPipes", "CoreAdvancedAssets"})
+public class ItemPipesTest {
+    @In
     private WorldProvider worldProvider;
+    @In
     private BlockManager blockManager;
+    @In
     private BlockEntityRegistry blockEntityRegistry;
+    @In
     private EntityManager entityManager;
+    @In
     private PipeSystem pipeSystem;
+    @In
     private Time time;
+    @In
+    private ModuleTestingHelper helper;
 
-    @Override
-    public Set<String> getDependencies() {
-        Set<String> modules = Sets.newHashSet();
-        modules.add("engine");
-        modules.add("CoreAdvancedAssets");
-        modules.add("SegmentedPaths");
-        modules.add("ItemPipes");
-        return modules;
-    }
+    private BlockFamily itemPipesBlockFamily;
+    private BlockFamily chestFamily;
+    private Block airBlock;
 
-    @Before
+    @BeforeEach
     public void initialize() {
-        worldProvider = getHostContext().get(WorldProvider.class);
-        blockManager = getHostContext().get(BlockManager.class);
-        blockEntityRegistry = getHostContext().get(BlockEntityRegistry.class);
-        entityManager = getHostContext().get(EntityManager.class);
-        pipeSystem = getHostContext().get(PipeSystem.class);
-        time = getHostContext().get(Time.class);
+        airBlock = blockManager.getBlock("engine:air");
+        itemPipesBlockFamily = blockManager.getBlockFamily("ItemPipes:basicPipe");
+        chestFamily = blockManager.getBlockFamily("CoreAdvancedAssets:Chest.LEFT");
+
+        BlockRegion region = new BlockRegion(0, 0, 0).expand(5, 5, 5);
+        for (Vector3ic pos : region) {
+            helper.forceAndWaitForGeneration(JomlUtil.from(pos));
+            worldProvider.setBlock(pos, airBlock);
+        }
     }
 
     @Test
     public void connectionTest() {
-        placeBlock(Vector3i.zero(), "ItemPipes:basicPipe");
-        assertEquals(0, getConn(Vector3i.zero()));
+        placePipe(new Vector3i());
+        Assertions.assertEquals(0, getConn(new Vector3i()));
```

Comment:
:raising_hand: This is the only part of this I'm asking for a change, and it's a small one:

As a reader, it's not always clear to me when things like `new Vecor3i()` are being passed just because the API demands a value there vs when they are meaningful for the test.

The change from `Vector3i.zero()` to `new Vector3i()` makes those doubts bigger: the `zero` helped me see that at least these are being passed equivalent values.

but I if I understand correctly, it is important that the argument to `placePipe` and the one to `getConn` have the same value.

I suggest factoring that argument out to a local variable `pipeLocation` or something.

That would change what's going on here slightly: you'd be passing the _same instance_ to both instead of _different instances with the same value._ Not sure if that matters to the intent of the test; it doesn't read to me like it should.

(maybe making it a `Vector3ic` would provide a hint that it's not changed by the methods its passed to?)

Reply:
I can concur with the sort of raised eyebrow I get from seeing stuff like `new Vector3i()` passed into something seemingly arbitrarily, plus it makes me wonder if we're actually creating a new object, while some constant such as `Vector3i.zero()` immediately makes me go "Oh, okay"

Reply:
That change should be simple enough, and in particular in the test cases we can be more explicit about `new Vector3i(0, 0, 0)` vs. `new Vector3i()`. Actually, `Vector3i.zero()` (teramath) is a method that creates a new vector instance for each call.

To test with both, passing the same argument to both methods and passing different arguments representing the same value, we could test with `pipeLocation` and `new Vector3i(pipeLocation)`. Maybe this would be a good property to test in more cases 🤔